File: java-backend/src/main/java/com/seekho/api/mapper/CampusLifeMapper.java
package com.seekho.api.mapper;

import com.seekho.api.dto.CampusLifeDto;
import com.seekho.api.entity.CampusLife;

public class CampusLifeMapper {

    public static CampusLifeDto toDto(CampusLife entity) {
        return new CampusLifeDto(
                entity.getId(),
                entity.getTitle(),
                entity.getDescription(),
                entity.getImageUrl()
        );
    }

    public static CampusLife toEntity(CampusLifeDto dto) {
        return new CampusLife(
                dto.getId(),
                dto.getTitle(),
                dto.getDescription(),
                dto.getImageUrl()
        );
    }
}

File: dotnet-backend/Mappers/CampusLifeMapper.cs
using dotnet_backend.Models;
using dotnet_backend.DTOs;

namespace dotnet_backend.Mappers
{
    public static class CampusLifeMapper
    {
        public static CampusLifeDto ToDto(CampusLife entity)
        {
            if (entity == null) return null;

            return new CampusLifeDto
            {
                CampusLifeId = entity.CampusLifeId,
                PhotoUrl = entity.PhotoUrl,
                Description = entity.Description
            };
        }

        public static CampusLife ToEntity(CampusLifeDto dto)
        {
            if (dto == null) return null;

            return new CampusLife
            {
                // Do NOT set CampusLifeId - it's auto-incremented by DB
                PhotoUrl = dto.PhotoUrl,
                Description = dto.Description
            };
        }
    }
}
